src/AccountManager.mo

```diff
@@ -491,40 +497,95 @@ module {
       };
     };
 
-    /// Initiates a withdrawal by transferring tokens to another account.
-    /// Returns ICRC1 transaction index and amount of transferred tokens (fee excluded).
-    public func withdraw(to : ICRC1.Account, amount : Nat) : async* WithdrawResponse {
-
-      totalWithdrawn_ += amount;
-
-      let callResult = await* processWithdrawTransfer(to, amount);
+    func handleWithdrawal(
+      from_subaccount : ?Blob,
+      to : ICRC1.Account,
+      amount : Nat,
+      onSuccess : (txIdx : Nat) -> async* WithdrawResponse,
+      onError : (err : ProcessWithdrawTransferError) -> async* WithdrawResponse,
```

Comment:
I wonder if we really need these. I understand we do the second transfer in onSuccess but what if we just return success/error and then on the level above do two sequential calls to handleWithdrawal? Doesn't that work equally well? I think it would be nice to read.